### PR TITLE
Add ok route, pass 418 cobspec tests

### PR DESCRIPTION
**Pull request by td-extreme**

### CODE REVIEW

File: spec/httpserver/handlers/ok_spec.clj [hunk 1]
```diff
@@ -0,0 +1,11 @@
+(ns httpserver.ok-spec
+  (:require  [speclj.core :refer :all]
```
Comment (mikeknep) [inline spec/httpserver/handlers/ok_spec.clj]: extra space here

File: spec/httpserver/handlers/ok_spec.clj [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+            [httpserver.handlers.ok :as ok]))
+
+(import '(com.td.HttpServer HttpResponse))
+
+(describe "teapotHandler"
```
Comment (mikeknep) [inline spec/httpserver/handlers/ok_spec.clj]: copy-pasta, this test is for the ok handler

File: spec/httpserver/handlers/ok_spec.clj [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+  (it "returns an HttpRequest with code 200"
+    (let  [testOk  (ok/new-ok-handler)
```
Comment (mikeknep) [inline spec/httpserver/handlers/ok_spec.clj]: extra space, and should be kebab cased, and also `test-ok` is kind of a weird variable name, why not `ok-handler`?

File: src/httpserver/core.clj [hunk 4]
```diff
@@ -10,6 +11,10 @@
 (defn -main [& args]
   (let [{:keys [options]} (arguments/build-map args)]
     (let [server (HttpServer. (:port options) (:directory options))
-          teapot-handler (teapot/new-teapot-handler)]
-      (.addRoute server "GET" "/teapot" teapot-handler)
+          teapot-handler (teapot/new-teapot-handler)
+          ok-handler (ok/new-ok-handler)
+          ]
```
Comment (mikeknep) [inline src/httpserver/core.clj]: Just a style thing—the closing `]` for `let` bindings is typically on the same line as the final binding. Also, no line empty line break between the bindings and the next forms (line below this)

File: src/httpserver/handlers/ok.clj [hunk 5]
```diff
@@ -0,0 +1,12 @@
+(ns httpserver.handlers.ok)
+
+(import '(com.td.HttpServer HttpResponse))
+(import '(com.td.HttpServer IHandler))
+
+(deftype ok-handler []
+  com.td.HttpServer.IHandler
+  (generateResponse [this]
+    (doto (HttpResponse.))))
```
Comment (mikeknep) [inline src/httpserver/handlers/ok.clj]: Huh? I'm confused. First of all, `doto` takes a data structure and a variable number of functions to run with it in the first argument spot. Check out the [documentation](https://clojuredocs.org/clojure.core/doto). What you have above is no different from simply `(HttpResponse.)`.

Second, uh, how does this work? Does `(HttpResponse.)` return a Response with 200 pre-set as the status?

Comment (mikeknep) [inline src/httpserver/handlers/ok.clj]: Also, a forewarning: careful with `doto`. It is designed for stateful objects, which are generally frowned upon in Clojure (but make sense for Java interop stuff). So, if your followup forms don't mutate the initial value `x`, you will just get that original, immutable value back. Contrast this with Clojure's threading macro, `->`, which applies the return value of each function to the next, and returns that final return value:

``` clojure
(doto 1 inc inc inc)  ;; 1
(-> 1 inc inc inc)    ;; 4
```

Comment (td-extreme) [inline src/httpserver/handlers/ok.clj]: Yeah, right now the default code for an HttpResponse is 200.

File: spec/httpserver/handlers/ok_spec.clj [hunk 6]
```diff
@@ -0,0 +1,11 @@
+(ns httpserver.ok-spec
+  (:require [speclj.core :refer :all]
+            [httpserver.handlers.ok :as ok]))
+
+(import '(com.td.HttpServer HttpResponse))
+
+(describe "OkHandler"
+  (it "returns an HttpRequest with code 200"
+    (let [ok-handler (ok/new-ok-handler)
+           response (.generateResponse ok-handler)]
```
Comment (mikeknep) [inline spec/httpserver/handlers/ok_spec.clj]: one last spacing issue here

Comment (mikeknep) [inline spec/httpserver/handlers/ok_spec.clj]: FYI, if you highlight a block of code in vim visual mode and press `=` it should fix all indentation issues

**Comment by mikeknep** [comment]
One final spacing issue and then :ship:

**Pull request merged**